modules/local/prelude/main.nf

```diff
@@ -0,0 +1,335 @@
+        )
+    }
+
+    n_channels = xml.'**'.findAll {
+            node -> node.name() == 'Pixels' && node.@SizeC != ''
+        }
+        .collect { node -> node.@SizeC.toInteger() }
+
+    if (n_channels == null || n_channels.toSet().size() != 1 || n_channels[0] == 0) {
+        output_xml.add(
+            ["SizeC", n_channels.toString(), "Consistent > 0 numbers", cross]
+        )
+    }
+    else {
+        output_xml.add(
+            ["SizeC", n_channels.toString(), "Consistent > 0 numbers", check]
+        )
+    }
+
+    size_units = xml.'**'.findAll {
+            node -> node.name() == 'Pixels' && node.@PhysicalSizeXUnit != '' && node.@PhysicalSizeYUnit != ''
+        }
+        .collect { node -> [node.@PhysicalSizeXUnit.toString(), node.@PhysicalSizeYUnit.toString()] }
+
+    if (size_units == null || size_units.toSet().size() != 1 || size_units.flatten().toSet().size() != 1 ||
+        size_units.flatten().toSet()[0] == null ||
+        !(size_units.flatten().toSet()[0] in ["mm", "cm", "um", "µm", "reference_frame"])) {
+        output_xml.add(
+            ["PhysicalSizeXUnit|PhysicalSizeYUnit", size_units.toString(), "Consistent units (mm, cm, um, µm, reference_frame)", cross]
+        )
+    }
+    else {
+        output_xml.add(
+            ["PhysicalSizeXUnit|PhysicalSizeYUnit", size_units.toString(), "Consistent units (mm, cm, um, µm, reference_frame)", check]
+        )
+    }
+
+    pixel_datatype = xml.'**'.findAll {
+            node -> node.name() == 'Pixels' && node.@Type
+        }
+        .collect { node -> node.@Type.toString() }
+
+    if (pixel_datatype == null || pixel_datatype.toSet().size() != 1 ||
+        !(pixel_datatype.toSet()[0] ==~ /[u]?int(8|16|32)|float|double/)  // There are more bit|complex|double-complex
+        ) {
+        output_xml.add(
+            ["Type", pixel_datatype.toString(), "Consistent valid datatypes (uint8, float16...)", cross]
+        )
+    }
+    else {
+        output_xml.add(
+            ["Type", pixel_datatype.toString(), "Consistent valid datatypes (uint8, float16...)", check]
+        )
+    }
+
+    exposure_time = xml.'**'.findAll {
+            node -> node.name() == 'Plane'
+        }
+        .collect {
+            node ->
+                [
+                    node.@ExposureTime.toDouble(),
+                    node.@ExposureTimeUnit.toString()
+                ]
+        }
+        .toSet()
+
+    if (exposure_time == null || exposure_time.size() != 1 || exposure_time[0] == null || exposure_time[1] == null || exposure_time[1] == "") {
+        output_xml.add(
+            ["ExposureTime|ExposureTimeUnit", exposure_time.toString(),
+            "Consistent valid exposure time and units", cross]
+        )
+    }
+    else {
+        output_xml.add(
+            ["ExposureTime|ExposureTimeUnit", exposure_time.toString(),
+            "Consistent valid exposure time and units", check]
+        )
+    }
+
+    output_file_xml = prefix + "_${meta.id}_xml_mqc.tsv"
```

Comment:
Update as per the previous comment, e.g. `prefix + "_xml_mqc.tsv"` or `prefix + "_summary_xml_mqc.tsv"` or similar.